presto-hive/src/main/java/io/prestosql/plugin/hive/s3select/S3SelectCsvRecordReader.java

```diff
@@ -55,8 +47,11 @@ public S3SelectCsvRecordReader(
     }
 
     @Override
-    public SelectObjectContentRequest buildSelectObjectRequest(Properties schema, String query, Path path)
+    public SelectObjectContentRequest buildSelectObjectRequest(Properties schema, String query, Path path, Configuration configuration)
     {
+        HiveS3Config defaults = new HiveS3Config();
+        String s3CsvComments = configuration.get(S3_CSV_COMMENTS, defaults.getS3CsvComments());
```

Comment:
Since this CSV comments config is only used by S3Select, we can limit it to the S3Select code -- no need to pass it through Hadoop config:

1. Change `PrestoS3ClientFactory` to take the new `HiveS3SelectConfig` as mentioned in the other comment.
2. Store `csvComments` from the config in `PrestoS3ClientFactory`.
3. Add a method `getCsvComments()` to `PrestoS3ClientFactory`and use that in the `S3SelectCsvRecordReader` constructor to fetch the value.